### PR TITLE
Set 'Setup Halted' event instead of 'Setup Failed' for exit code 3

### PR DESCRIPTION
**Pull request by nathanleclaire**
cc @dgageot @jeanlaurent @FrenchBen @jeffdm  FYI

Requires: https://github.com/docker/machine/pull/2979

Signed-off-by: Nathan LeClaire <nathan.leclaire@gmail.com>

### CODE REVIEW

File: src/utils/Util.js [hunk 1]
```diff
@@ -8,12 +8,23 @@ const remote = electron.remote;
...
 
+// Docker Machine exits with 3 to differentiate pre-create check failures (e.g.
+// virtualization isn't enabled) from normal errors during create (exit code
+// 1).
+const precreateCheckExitCode = 3;
+
 module.exports = {
+  errPrecreateCheckFailure:  'Pre-create check failed: ',
   execFile: function (args, options) {
+    var that = this;
```
Comment (jeffdm) [inline src/utils/Util.js]: Should be unnecessary given that we use fat arrow functions. I.e. the  `=> {`  syntax.

Comment (FrenchBen) [inline src/utils/Util.js]: ES6 - You shouldn't need to do 'that' for 'this' :)

File: src/utils/Util.js [hunk 2]
```diff
@@ -8,12 +8,23 @@ const remote = electron.remote;
...
     return new Promise((resolve, reject) => {
       child_process.execFile(args[0], args.slice(1), options, (error, stdout) => {
         if (error) {
-          reject(new Error('Encountered an error: ' + error));
+          if (error.code === precreateCheckExitCode) {
+            reject(new Error(that.errPrecreateCheckFailure + error));
+          } else {
+            reject(new Error('Encountered an error: ' + error));
```
Comment (jeffdm) [inline src/utils/Util.js]: Simple return `error` here via `reject(error)`. Then check `if (error.code && error.code === 3)` in the `SetupUtil` code.

**Comment by jeffdm** [comment]
Great! Thanks @nathanleclaire . Some small comments on my end

File: src/utils/Util.js (hunk 1 above)
Comment (nathanleclaire) [inline src/utils/Util.js]: :heart_eyes:

**Comment by nathanleclaire** [comment]
Revised my PR to not be such a ES6 / promise n00b.  PTAL

**Pull request merged**